### ISSUE TITLE
Stale NFS file handle

### ISSUE THREAD

**Issue by alexyao2015** [kubernetes-csi/csi-driver-smb#419]
**What happened**:

When the smb server is terminated while still mounted to pods, the csi driver node does not umount the stale mount and create a new mount. This prevents new pods from starting even when the server is back online. Instead an error results. This can be resolved manually by restarting all the `csi-smb-node` pods. 

```
Error: failed to generate container "88d821046af27afe3710f0ffc413529b7eb46f2844156cfb154099dc94d75984"
spec: failed to generate spec: failed to stat "/var/lib/kubelet/pods/7244575c-5464-4faa-98f1-a80a2366287f/volumes/kubernetes.io~csi/smb-pv/mount": 
stat /var/lib/kubelet/pods/7244575c-5464-4faa-98f1-a80a2366287f/volumes/kubernetes.io~csi/smb-pv/mount: stale NFS file handle
```

**What you expected to happen**:

Not have to restart the `csi-smb-node` pod manually to mount.

**How to reproduce it**:

1. Start pod with smb mount.
2. Kill smb server.
3. Restart pod (shouldn't attach because server is down).
4. Start smb server.
5. Observe pod is never able to start even when server is back up.

**Anything else we need to know?**:

This may be related to #164, but it is showing a different error message.

**Environment**:
- CSI Driver version: v1.5.0
- Kubernetes version (use `kubectl version`): v1.20
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by andyzhangx** [comment]
the error shows `stale NFS file handle` while this is smb server

**Comment by alexyao2015** [comment]
Yes, this is correct. It's perplexing why it's showing stale nfs when it is smb.

**Comment by phxescap** [comment]
I have the same issue using the smb csi driver. For some reason the error "stale NFS file handle" only affects Linux containers trying to mount the smb share. Windows containers show no error message and continue to mount the share successfully after smb server restart.

**Comment by rmoreas** [comment]
Restarting csi-smb-node pods didn't recover the stale mount for us either.

Is there another way to recover from this issue without rebooting the node?

**Comment by ThomVivet** [comment]
From time to time we have the same `stale NFS file handle` error message when restarting a pod that use SMB mounts.
I found a workaround by a scaling down/up the application... I know it's not good for production environments.
Also I noticed that the same share is mounted twice:
- once into the pod: /var/lib/kubelet/pods/xxx/volumes/kubernetes.io~csi/xxx/mount 
- and another into the driver: /var/lib/kubelet/plugins/kubernetes.io/csi/smb.csi.k8s.io/xxx/globalmount

**Comment by DarkFM** [comment]
@andyzhangx 
Is it possible this was resolved in this Kubernetes v1.22.5? or is that unrelated?
https://github.com/kubernetes/kubernetes/pull/107039

**Comment by andyzhangx** [comment]
> @andyzhangx Is it possible this was resolved in Kubernetes v1.22.5? or is that unrelated? [kubernetes/kubernetes#107039](https://github.com/kubernetes/kubernetes/pull/107039)

@DarkFM I think that's related.

**Comment by iMartyn** [comment]
I can confirm that this still isn't fixed - running 1.32.5 k8s and the latest csi-driver-smb, and still regularly if the server goes away, I have to reboot the node that they were on, or I get the same old "Stale NFS Handle" and `ContainerErrorCreating` pods.

```
  Normal   Pulled                  33s                    kubelet                  Successfully pulled image "ghcr.io/hotio/sonarr:release-4.0.13.2932" in 556ms (556ms including waiting). Image size: 108531273 bytes.
  Warning  Failed                  33s                    kubelet                  Error: failed to generate container "b3835a07f1927d8b1600b914aabfc5c65169c6c1ef23a4e06d87aeb5f2b70a75" spec: failed to apply OCI options: failed to stat "/var/lib/kubelet/pods/a70b4271-ce52-4b99-93d3-27c2a2a5720e/volumes/kubernetes.io~csi/pvc-a7eb6f61-4806-49e8-843e-2ea463c057ce/mount": stat /var/lib/kubelet/pods/a70b4271-ce52-4b99-93d3-27c2a2a5720e/volumes/kubernetes.io~csi/pvc-a7eb6f61-4806-49e8-843e-2ea463c057ce/mount: stale NFS file handle
  Normal   Pulled                  32s                    kubelet                  Successfully pulled image "ghcr.io/hotio/sonarr:release-4.0.13.2932" in 708ms (708ms including waiting). Image size: 108531273 bytes.
  Warning  Failed                  32s                    kubelet                  Error: failed to generate container "602554a98f1b572f96b6c763cb427da12d3cef20e25384dc6b73feed0b74997c" spec: failed to apply OCI options: failed to stat "/var/lib/kubelet/pods/a70b4271-ce52-4b99-93d3-27c2a2a5720e/volumes/kubernetes.io~csi/pvc-a7eb6f61-4806-49e8-843e-2ea463c057ce/mount": stat /var/lib/kubelet/pods/a70b4271-ce52-4b99-93d3-27c2a2a5720e/volumes/kubernetes.io~csi/pvc-a7eb6f61-4806-49e8-843e-2ea463c057ce/mount: stale NFS file handle
  Normal   Pulled                  17s                    kubelet                  Successfully pulled image "ghcr.io/hotio/sonarr:release-4.0.13.2932" in 793ms (794ms including waiting). Image size: 108531273 bytes.
  Warning  Failed                  17s                    kubelet                  Error: failed to generate container "db4bb7e795b15383e3775dc2801089087ab1ac41e8f70f5d57ff7d56578c9283" spec: failed to apply OCI options: failed to stat "/var/lib/kubelet/pods/a70b4271-ce52-4b99-93d3-27c2a2a5720e/volumes/kubernetes.io~csi/pvc-a7eb6f61-4806-49e8-843e-2ea463c057ce/mount": stat /var/lib/kubelet/pods/a70b4271-ce52-4b99-93d3-27c2a2a5720e/volumes/kubernetes.io~csi/pvc-a7eb6f61-4806-49e8-843e-2ea463c057ce/mount: stale NFS file handle
  Normal   Pulling                 3s (x4 over 34s)       kubelet                  Pulling image "ghcr.io/hotio/sonarr:release-4.0.13.2932"
  Normal   Pulled                  2s                     kubelet                  Successfully pulled image "ghcr.io/hotio/sonarr:release-4.0.13.2932" in 550ms (550ms including waiting). Image size: 108531273 bytes.
  Warning  Failed                  2s                     kubelet                  Error: failed to generate container "b740ab139f3a640045ea4f343a0f3f38208dcbc38047ca263dfcd0bd8b844427" spec: failed to apply OCI options: failed to stat "/var/lib/kubelet/pods/a70b4271-ce52-4b99-93d3-27c2a2a5720e/volumes/kubernetes.io~csi/pvc-a7eb6f61-4806-49e8-843e-2ea463c057ce/mount": stat /var/lib/kubelet/pods/a70b4271-ce52-4b99-93d3-27c2a2a5720e/volumes/kubernetes.io~csi/pvc-a7eb6f61-4806-49e8-843e-2ea463c057ce/mount: stale NFS file handle
```
